internal/integration/cleanup_queues_test.go

```diff
@@ -0,0 +1,145 @@
+package integration_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/buildkite/agent-stack-k8s/v2/internal/integration/api"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCleanupOrphanedQueues(t *testing.T) {
```

Comment:
So my PR (#792) added similar functionality to the `CleanupOrphanedResources` test, because we can't delete queues that have jobs in them, so there's a light dependency between cancelling jobs (which happens as part of pipeline cleanup) and deleting queues.

because run order is randomised, this means that sometimes this "test" will run in parallel with `TestCleanupOrphanedResources`, which might lead to contention. i think it's best that we keep all of the cleanup logic in one test so that we can control order more easily.